### ISSUE TITLE
Intent to implement: use GitHub Actions for tests

### ISSUE THREAD

**Issue by merlinnot** [grpc/grpc-node#1170]
### Is your feature request related to a problem? Please describe.

I noticed that CI runs must be triggered manually by contributors. I was wondering if we can make this process more optimal by utilizing GitHub Actions, which support Linux, macOS and Windows too.

### Describe the solution you'd like

I'd be willing to work on a PR that introduces CI workflows using GitHub Actions, so these can be triggered automatically for every PR.

### Describe alternatives you've considered

There are many more CIs available, but nowadays GitHub Actions seems to be a to-go solution for any GitHub repository because of great functionalities and very tight integration.

### Additional context

**Comment by murgatroid99** [comment]
Because of the scale of our testing and the amount of computing resources required, we feel that we're better off using our Google-maintained CI infrastructure than what GitHub provides.

In addition, we would want to restrict who can initiate CI runs in either case to prevent people from abusing our computing resources to run their own computationally expensive programs.

**Comment by merlinnot** [comment]
Yes, for your own infrastructure it's definitely better to have control, that's why I was thinking about some external CIs that have protections against these kind of things built-in.

That's a shame, but I'll just try to always make sure tests are passing locally.

Do you think it would be possible to use GitHub Actions for tasks like `gts check`, so at least this part is more contributor-friendly?
